src/Tests/With_UTF8_BOM.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.IO;
+using System.Text;
+using NServiceBus.MessageInterfaces.MessageMapper.Reflection;
+using NServiceBus.Newtonsoft.Json;
+using NUnit.Framework;
+
+[TestFixture]
+public class With_UTF8_BOM
+{
+
+    [Test]
+    public void Run()
```

Comment:
Let's rename this to `Should_handle_message_with_UTF8_BOM`

Reply:
I'd also like to see a Should_handle_message_without_UTF8_BOM test.

Reply:
Added.